File: app/src/main/java/seigneur/gauvain/mycourt/data/local/dao/TokenDao.java
package seigneur.gauvain.mycourt.data.local.dao;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;

import io.reactivex.Maybe;
import seigneur.gauvain.mycourt.data.model.Token;

@Dao
public interface TokenDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    long insertToken(Token token);

    @Query("SELECT * FROM token")
    Maybe<Token> getToken();

}

File: app/src/main/java/seigneur/gauvain/mycourt/data/local/dao/TokenDao.kt
package seigneur.gauvain.mycourt.data.local.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query

import io.reactivex.Maybe
import seigneur.gauvain.mycourt.data.model.Token

@Dao
interface TokenDao {

    @get:Query("SELECT * FROM token")
    val token: Maybe<Token>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertToken(token: Token): Long

}
